Source/WebKit/WebProcess/WebPage/WebCookieJar.h

```diff
@@ -76,9 +80,18 @@ class WebCookieJar final : public WebCore::CookieJar {
     bool remoteCookiesEnabledSync(WebCore::Document&) const;
     void clearCacheForHost(const String&) final;
     bool isEligibleForCache(WebFrame&, const URL& firstPartyForCookies, const URL& resourceURL) const;
+    String cookiesInPartitionedCookieStorage(const WebCore::Document&, const URL&, const WebCore::SameSiteInfo&) const;
+    void setCookiesInPartitionedCookieStorage(const WebCore::Document&, const URL&, const WebCore::SameSiteInfo&, const String& cookieString);
+#if PLATFORM(COCOA)
+    NSHTTPCookieStorage* ensurePartitionedCookieStorage();
+#endif
 
     mutable WebCookieCache m_cache;
     HashMap<String, WeakHashSet<WebCore::CookieChangeListener>> m_changeListeners;
+
+#if PLATFORM(COCOA)
+    RetainPtr<NSHTTPCookieStorage> m_partitionedCookieStorage;
```

Comment:
I think this needs a better name and maybe a comment.

Reply:
Have you thought about the lifetime of these cookies? Seems this storage never gets cleared (or removed from). It also seems it is page wide and pages can get navigated from one domain to another? Do we care about the fact that cookies for a given domain stay in the process basically forever, even after you're done visiting this domain?

Reply:
I am not too concerned about the lifetime because the cookies are small, non-persistent and there is only one quirked domain now, and we swap process on navigation. Also the page could navigate back to the domain so we might want to keep them.

Reply:
I believe these cookies exist for the lifetime of the WebContent process, not the Network Process, correct? If so, I agree the lifetime should not be a concern.

Reply:
Yes, this is only in the memory of web process and not used for network requests.